x-pack/plugin/ilm/qa/multi-cluster/src/test/java/org/elasticsearch/xpack/ilm/CCRIndexLifecycleIT.java

```diff
@@ -839,4 +965,32 @@ private static String getShrinkIndexName(RestClient client, String originalIndex
             : "lifecycle execution state must contain the target shrink index name for index [" + originalIndex + "]";
         return shrunkenIndexName[0];
     }
+
+    private static Map<String, Object> explainIndex(RestClient client, String indexName) throws IOException {
+        RequestOptions consumeWarningsOptions = RequestOptions.DEFAULT.toBuilder()
+            .setWarningsHandler(warnings -> warnings.isEmpty() == false && List.of("""
+                [indices.lifecycle.rollover.only_if_has_documents] setting was deprecated in Elasticsearch \
+                and will be removed in a future release. \
+                See the deprecation documentation for the next major version.""").equals(warnings) == false)
+            .build();
```

Comment:
Hm, I was surprised to see this warning. Is this being triggered by this test?

Reply:
No worries, it happens to all of us. I will follow up with the original code, I am curious why this was there if this isn't triggered.